recipes/mp-units/2.0.0/conanfile.py

```diff
@@ -0,0 +1,158 @@
+import os
+
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get, rmdir
+from conan.tools.scm import Version
+
+required_conan_version = ">=1.59.0"
+
+
+class MPUnitsConan(ConanFile):
+    name = "mp-units"
+    homepage = "https://github.com/mpusz/mp-units"
+    description = "A Physical Quantities and Units library for C++"
+    topics = (
+        "units",
+        "dimensions",
+        "quantities",
+        "dimensional-analysis",
+        "physical-quantities",
+        "physical-units",
+        "system-of-units",
+        "system-of-quantities",
+        "isq",
+        "si",
+        "library",
+        "quantity-manipulation",
+    )
+    license = "MIT"
+    url = "https://github.com/conan-io/conan-center-index"
+    settings = "os", "arch", "compiler", "build_type"
+    tool_requires = "cmake/[>=3.19 <4]"
+    package_type = "header-library"
+    no_copy_source = True
+
+    @property
+    def _min_cppstd(self):
+        return "20"
+
+    @property
+    def _minimum_compilers_version(self):
+        return {"gcc": "11", "clang": "16"}
```

Comment:
Do you know the min required versions for `msvc` and `apple-clang`?

Reply:
Even though we require "only" C++20 support MSVC is too buggy to support that.

Reply:
Regarding apple-clang, for now we do not have enough experience to know if it works on the latest compiler version already. I have other priorities now and did not have time to investigate if we can make it work. But taking into account that mp-units 2.0 barely compiles (with many hacks) on clang-16, we expect it will take a while for apple-clang to catch up.

Reply:
Wondering just now if we should disallow apple-clang and msvc builds for now then?